lib/ansible/modules/cloud/vmware/vmware_vswitch.py

```diff
@@ -129,55 +253,121 @@
+                vswitch_failure['fullDuplex'] = available_vswitch.spec.policy.nicTeaming.failureCriteria.fullDuplex
+                vswitch_failure['checkErrorPercent'] = \
+                    available_vswitch.spec.policy.nicTeaming.failureCriteria.checkErrorPercent
+                vswitch_failure['percentage'] = available_vswitch.spec.policy.nicTeaming.failureCriteria.percentage
+                vswitch_failure['checkBeacon'] = available_vswitch.spec.policy.nicTeaming.failureCriteria.checkBeacon
+                vswitch_ts = dict()
+                vswitch_ts['enabled'] = available_vswitch.spec.policy.shapingPolicy.enabled
+                vswitch_ts['averageBandwidth'] = available_vswitch.spec.policy.shapingPolicy.averageBandwidth
+                vswitch_ts['peakBandwidth'] = available_vswitch.spec.policy.shapingPolicy.peakBandwidth
+                vswitch_ts['burstSize'] = available_vswitch.spec.policy.shapingPolicy.burstSize
+                vswitch_offload = dict()
+                vswitch_offload['csumOffload'] = available_vswitch.spec.policy.offloadPolicy.csumOffload
+                vswitch_offload['tcpSegmentation'] = available_vswitch.spec.policy.offloadPolicy.tcpSegmentation
+                vswitch_offload['zeroCopyXmit'] = available_vswitch.spec.policy.offloadPolicy.zeroCopyXmit
                 used_pnic = []
                 for pnic in available_vswitch.pnic:
                     # vSwitch contains all PNICs as string in format of 'key-vim.host.PhysicalNic-vmnic0'
                     m_pnic = pnic.split("-", 3)[-1]
                     used_pnic.append(m_pnic)
-                self.available_vswitches[available_vswitch.name] = dict(pnic=used_pnic,
-                                                                        mtu=available_vswitch.mtu,
-                                                                        num_ports=available_vswitch.spec.numPorts,
-                                                                        )
-            for desired_pnic in self.nics:
-                if desired_pnic not in available_pnic:
-                    # Check if pnic does not exists
-                    self.module.fail_json(msg="Specified Physical NIC '%s' does not"
-                                              " exists on given ESXi '%s'." % (desired_pnic,
-                                                                               self.host_system.name))
-                for vswitch in self.available_vswitches:
-                    if desired_pnic in self.available_vswitches[vswitch]['pnic'] and vswitch != self.switch:
-                        # Check if pnic is already part of some other vSwitch
-                        self.module.fail_json(msg="Specified Physical NIC '%s' is already used"
-                                                  " by vSwitch '%s'." % (desired_pnic, vswitch))
+                self.available_vswitches[available_vswitch.name] = dict(
```

Comment:
Use dict literal
```suggestion
                self.available_vswitches[available_vswitch.name] = {
```

Reply:
This doesn't work. I get E225 errors when I do this...

Reply:
It does work. This error is about putting whitespaces https://lintlyci.github.io/Flake8Rules/rules/E225.html, which probably means that you introduced style offences along the way.